### PR TITLE
refactored sample code test program and fixed a bug

### PR DESCRIPTION
**Pull request by suoych**
"test=develop"
refactored sample code: deleted redundant code and renamed some variable, made the code easier to understand.
fixed a bug: now the program could print all the error and exit with return code 1

### CODE REVIEW

File: tools/sampcd_processor.py [hunk 1]
```diff
@@ -20,14 +20,13 @@
...
+usage: python sample_test.py {arg1} 
 arg1: the first arg defined running in gpu version or cpu version
-arg2: the second arg defined testing python2 or python3 
 
 for example, you can run cpu version python2 testing like this:
-    
-    python sample_test.py cpu 2
-    
+
+    python sample_test.py cpu
```
Comment (shippingwang) [inline tools/sampcd_processor.py]: typo -> sampcd_processor.py

Comment (suoych) [reply tools/sampcd_processor.py]: fixed, thanks.

**Review by shippingwang — APPROVED** [review]
Good job!

**Comment by suoych** [comment]
> Good job!

thanks

**Pull request merged**